rotkehlchen/balances/historical.py

```diff
@@ -78,6 +79,33 @@ def get_asset_balance(self, asset: Asset, timestamp: Timestamp) -> HistoricalBal
 
         return {'amount': amounts[asset], 'price': price}
 
+    def get_collection_amounts(
+            self,
+            collection_id: int,
+            from_ts: Timestamp,
+            to_ts: Timestamp,
+    ) -> tuple[dict[Timestamp, FVal], str | None]:
+        """Get historical balance amounts for all assets in a collection within the given time range.
+
+        Similar to get_asset_amounts() but for all assets in a collection. The amounts
+        of each asset in the collection are aggregated into a single timeline.
+
+        May raise:
+        - NotFoundError if no events exist for any asset in the collection in the specified time period.
+        """  # noqa: E501
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT asset FROM multiasset_mappings WHERE collection_id=?',
+                (collection_id,),
+            )
+            assets = tuple(Asset(row[0]) for row in cursor)
+
+        events, _ = self._get_events_and_currency(from_ts=from_ts, to_ts=to_ts, assets=assets)
+        if len(events) == 0:
+            raise NotFoundError(f'No historical data found for collection {collection_id} between {from_ts=} and {to_ts=}.')  # noqa: E501
```

Comment:
looks like this part can also be moved to the common logic? The only thing is that for the string you will have some check of `if len(assets) > 1`. I would change `get_asset_amounts` to `get_assets_amounts` and when you call it in rest.py just to `assets=(asset,)`. 

Then `get_collection_amounts` is just a wrapper that gets the assets by collection id